src/server-extension/resolvers/AdminResolver/index.ts

```diff
@@ -65,14 +103,48 @@ export class AdminResolver {
         args.commentsWeight,
         args.reactionsWeight,
         [args.joysteamTimestampSubWeight, args.ytTimestampSubWeight],
+        args.defaultChannelWeight,
       ],
       em
     )
     await videoRelevanceManager.updateVideoRelevanceValue(em, true)
     return { isApplied: true }
   }
 
-  @UseMiddleware(OperatorOnly)
+  @UseMiddleware(OperatorOnly(OperatorPermission.SET_CHANNEL_WEIGHTS))
+  @Mutation(() => [ChannelWeight])
+  async setChannelsWeights(@Args() { inputs }: SetChannelsWeightsArgs): Promise<ChannelWeight[]> {
+    const em = await this.em()
+
+    const results: ChannelWeight[] = []
+
```

Comment:
I was wondering that now that we have permissions in place it might happens that there can be more than one user with the `SET_CHANNEL_WEIGHTS` permission and they might try to call this mutation at the same time.
Should we consider locking for this (and possibly other mutations)?

Reply:
I get your point but I don't think it's really necessary to lock the records, since locks have their own overhead, and they put all the future transactions affecting the record in `idle` state. which could have a drastic effect on performance (e.g. If we use lock in this mutation, then all the read/write operations by other mappings/queries/operations will be blocked until this operation is completed). Also, this specific transaction is only about updating a single field of the record, and even if there is any concurrent transaction affecting the record, at worst the channel weight update would fail, which I think is fine (as you would see that in the mutation [response](https://github.com/Joystream/orion/pull/232#discussion_r1390901808)).

Another important thing is that maybe we want to review all the `pessimistic_write` lock usage in queries/mutations and see if it's really required.